packages/common/src/controllers/AccountController.ts

```diff
@@ -418,6 +418,12 @@ export default class AccountController {
 
     let pendingOffer: Offer | null = null;
 
+    if (!activeSubscription && !!retry && retry > 0) {
```

Comment:
I guess there is at least 1 edge case that this won't behave as intended for; if the viewer already has a subscription and is buying another one to access additional content. Would it be possible to pass an ID or something to the retry / wait recursion so we can keep trying until we see the expected new subscription?

Reply:
Hm, good suggestion. With the suggested background polling in mind, perhaps we can create a `checkSubscription(id)` method? We could use it from `useCheckAccess()`, to make sure that the correct subscription is awaited.